src/Shared/Guard.cs

```diff
@@ -19,12 +19,12 @@ namespace System.Runtime.CompilerServices
     [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false, Inherited = false)]
     internal sealed class CallerArgumentExpressionAttribute : Attribute
     {
-        public CallerArgumentExpressionAttribute(string parameterName)
+        public CallerArgumentExpressionAttribute(string? parameterName)
```

Comment:
Should doublecheck this. The original class doesn't have nullable. ([link](https://github.com/dotnet/runtime/blob/b14e2f5ef3008ab62bdbe8044f00fe948e20595e/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/CallerArgumentExpressionAttribute.cs#L12))

My understanding is that this should never be null.

Reply:
See also https://github.com/dotnet/runtime/blob/v9.0.0-rc.1.24431.7/src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Runtime/CompilerServices/CallerArgumentExpressionAttributeTests.cs (the same tests we have copied to this repository).

I am not against dropping support for null, but it is kind of break for what we have in dotnet repository.

Whatever we decide, the impact should be low. It is extremely internal class. Any recommendation based on this?

Reply:
Thanks. I think this changes is good.